lib/tables.c

```diff
@@ -4675,3 +4676,142 @@ table_collection_simplify(table_collection_t *self,
     simplifier_free(&simplifier);
     return ret;
 }
+
+
+/*******************
+ * Cleaning tables *
+ ******************/
+
+typedef struct {
+    site_table_t *sites;
+    mutation_table_t *mutations;
+    /* Map of old site IDs to new site IDs. */
+    site_id_t *site_id_map;
+} table_cleaner_t;
+
+static int table_cleaner_run(table_cleaner_t *self)
+{
+    // Remove any sites with duplicate positions, retaining only the *first*
+    // one. Assumes the tables have been sorted, throwing an error if not.
+
+    int ret = 0;
+    table_size_t j, k, site_j;
+    table_size_t as_length, as_offset;
+    table_size_t md_length, md_offset;
+    table_size_t num_input_sites;
+    double last_position, position;
+
+    num_input_sites = self->sites->num_rows;
+    site_j = 0; // the index of the next output row
+    last_position = -INFINITY;
```

Comment:
I usually use DBL_MIN, or -1 for this (presumably this is what the math.h is for?). -1 is the same thing really since we're not allowing negative positions.

Reply:
ah, thats better.

Reply:
Actually, it's `-DBL_MAX` (since `DBL_MIN` is the smallest *positive* float), and we need to include `float.h` for that one...

Reply:
I wonder whether -DBL_MAX is guaranteed to be representable in double.  That's a little tricky for signed integer types; INT8_MAX is 127 whereas INT8_MIN is -128.  So -INT8_MAX is correctly representable, but -INT8_MIN is not.  I'm not sure what the rules are for IEEE double, and whether -DBL_MAX is safe.  Surely a smaller negative value, like -1e1000 or something, would suffice?

Reply:
According to [this post](https://stackoverflow.com/questions/1153548/minimum-double-value-in-c-c) which references [ANSI C](http://www.csse.uwa.edu.au/programming/ansic-library.html#float) it is...

Reply:
although that link to ANSI C does not state it.

Reply:
I think -1 is probably fine. It's what I usually use for this requirement: any value < 0 will do.